sale_order_country_allowed_product/models/product_product.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2025 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.model
+    def _search(
+        self,
+        args,
+        offset=0,
+        limit=None,
+        order=None,
+        count=False,
+        access_rights_uid=None,
+    ):
+        if self.env.context.get(
+            "restrict_by_country", False
+        ) and not self.env.user.has_group(
+            "sale_order_country_allowed_product.ignore_country_sale"
+        ):
+            partner = self.env["res.partner"].search(
+                [("id", "=", self.env.context.get("restrict_by_country_partner_id"))],
+                limit=1,
+            )
+            if partner.country_id:
+                args += [
```

Comment:
I suggest using [expression.AND](https://github.com/odoo/odoo/blob/a71c0c7d1c3391beb1bf2939ae076a804a627dbd/addons/account/controllers/portal.py#L82) in order to ensure that domain is formed properly.